src/lib.rs

```diff
@@ -422,13 +422,14 @@ fn format_ast<F>(
     config: &Config,
     report: FormatReport,
     mut after_file: F,
-) -> Result<(FileMap, bool), io::Error>
+) -> Result<(FileMap, bool, bool), io::Error>
 where
     F: FnMut(&FileName, &mut String, &[(usize, usize)], &FormatReport) -> Result<bool, io::Error>,
 {
     let mut result = FileMap::new();
     // diff mode: check if any files are differing
     let mut has_diff = false;
+    let mut has_macro_rewrite_failure = false;
```

Comment:
@topecongiro @nrc 
Shouldn't `has_macro_rewrite_failure` be part of the `FormatReport` instead?
IMO That should be a new ErrorKind

Reply:
I figure the two will get merged in some way, so it doesn't really matter what we do right now